tests/components/hyperion/test_light.py

```diff
@@ -0,0 +1,430 @@
+
+    # By default the effect list is the 3 external sources + 'Solid'.
+    assert len(entity_state.attributes["effect_list"]) == 4
+
+    assert (
+        entity_state.attributes["supported_features"] == hyperion_light.SUPPORT_HYPERION
+    )
+
+
+async def test_light_async_turn_on(hass):
+    """Test turning the light on."""
+    client = create_mock_client()
+    await setup_entity(hass, client=client)
+
+    # On (=), 100% (=), solid (=), [255,255,255] (=)
+    client.async_send_set_color = CoroutineMock(return_value=True)
+    await hass.services.async_call(
+        DOMAIN, SERVICE_TURN_ON, {ATTR_ENTITY_ID: TEST_ENTITY_ID}, blocking=True
+    )
+
+    assert client.async_send_set_color.call_args == call(
+        **{
+            const.KEY_PRIORITY: TEST_PRIORITY,
+            const.KEY_COLOR: [255, 255, 255],
+            const.KEY_ORIGIN: hyperion_light.DEFAULT_ORIGIN,
+        }
+    )
+
+    # On (=), 50% (!), solid (=), [255,255,255] (=)
+    # ===
+    brightness = 128
+    client.async_send_set_color = CoroutineMock(return_value=True)
+    client.async_send_set_adjustment = CoroutineMock(return_value=True)
+    await hass.services.async_call(
+        DOMAIN,
+        SERVICE_TURN_ON,
+        {ATTR_ENTITY_ID: TEST_ENTITY_ID, ATTR_BRIGHTNESS: brightness},
+        blocking=True,
+    )
+
+    assert client.async_send_set_adjustment.call_args == call(
+        **{const.KEY_ADJUSTMENT: {const.KEY_BRIGHTNESS: 50}}
+    )
+    assert client.async_send_set_color.call_args == call(
+        **{
+            const.KEY_PRIORITY: TEST_PRIORITY,
+            const.KEY_COLOR: [255, 255, 255],
+            const.KEY_ORIGIN: hyperion_light.DEFAULT_ORIGIN,
+        }
+    )
+
+    # Simulate a state callback from Hyperion.
+    client.adjustment = [{const.KEY_BRIGHTNESS: 50}]
+    call_registered_callback(client, "adjustment-update")
+    entity_state = hass.states.get(TEST_ENTITY_ID)
+    assert entity_state.state == "on"
+    assert entity_state.attributes["brightness"] == brightness
+
+    # On (=), 50% (=), solid (=), [0,255,255] (!)
+    hs_color = (180.0, 100.0)
+    client.async_send_set_color = CoroutineMock(return_value=True)
+    await hass.services.async_call(
+        DOMAIN,
+        SERVICE_TURN_ON,
+        {ATTR_ENTITY_ID: TEST_ENTITY_ID, ATTR_HS_COLOR: hs_color},
+        blocking=True,
+    )
+
+    assert client.async_send_set_color.call_args == call(
+        **{
+            const.KEY_PRIORITY: TEST_PRIORITY,
+            const.KEY_COLOR: (0, 255, 255),
+            const.KEY_ORIGIN: hyperion_light.DEFAULT_ORIGIN,
+        }
+    )
+
+    # Simulate a state callback from Hyperion.
+    client.visible_priority = {
+        const.KEY_COMPONENTID: const.KEY_COMPONENTID_COLOR,
+        const.KEY_VALUE: {const.KEY_RGB: (0, 255, 255)},
+    }
+
+    call_registered_callback(client, "priorities-update")
+    entity_state = hass.states.get(TEST_ENTITY_ID)
+    assert entity_state.attributes["hs_color"] == hs_color
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6233107)